Fix flaky Kafka integration test

**TestKafkaStorage/GetTrace** fails and causes the build to fail. Update the test to be reliable. 


```
--- FAIL: TestKafkaStorage (6.18s)
    --- FAIL: TestKafkaStorage/GetTrace (3.02s)
    	integration_test.go:99: Waiting for storage backend to update documents, iteration 1 out of 30
    	integration_test.go:180: Trace was not found
    	. 
        .
        .
    	integration_test.go:99: Waiting for storage backend to update documents, iteration 30 out of 30
    	integration_test.go:180: Trace was not found
    	integration_test.go:180: Trace was not found
    	assertions.go:256: 
    			Error Trace:	integration_test.go:184
    			            				integration_test.go:377
    			Error:      	Should be true
    			Test:       	TestKafkaStorage/GetTrace
    	assertions.go:256: 
    			Error Trace:	domain_trace_compare_test.go:50
    			            				integration_test.go:185
    			            				integration_test.go:377
    			Error:      	Expected value not to be nil.
    			Test:       	TestKafkaStorage/GetTrace
```

--- a/scripts/travis/kafka-integration-test.sh
+++ b/scripts/travis/kafka-integration-test.sh
@@ -4,6 +4,14 @@ set -e
 
 docker pull spotify/kafka
 CID=$(docker run -d -p 2181:2181 -p 9092:9092 --env ADVERTISED_HOST=localhost --env ADVERTISED_PORT=9092 --rm spotify/kafka)
+
+# Guarantees no matter what happens, docker will remove the instance at the end.
+trap 'docker rm -f $CID 2>/dev/null' EXIT INT TERM
+
 export STORAGE=kafka
+while true; do
+    if nc -z localhost 9092; then
+        break
+    fi
+done
 make storage-integration-test
-docker kill $CID

